Scripts/Runtime/Data/Collections/DeferredNativeArray.cs

```diff
@@ -19,21 +19,31 @@ namespace Anvil.Unity.DOTS.Data
     /// </remarks>
     public unsafe struct DeferredNativeArrayScheduleInfo
     {
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
         internal void* SafetyHandlePtr
         {
             get;
         }
+#endif
 
         internal void* BufferPtr
         {
             get;
         }
 
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
```

Comment:
This is a bit awkward. 
Alternatively I could have duplicated the type in the different `#if` conditions but that didn't seem any better.

Ideally it would be best to not include the safety handle pointer in the struct and instead request it from the `DeferredNativeArray` instance but I don't know if that's feasible.